src/Raven.Server/Documents/Replication/Outgoing/OutgoingMigrationReplicationHandler.cs

```diff
@@ -52,6 +52,14 @@ protected override DynamicJsonValue GetInitialHandshakeRequest()
         {
             var request = base.GetInitialHandshakeRequest();
             request[nameof(ReplicationLatestEtagRequest.MigrationIndex)] = BucketMigrationNode.MigrationIndex;
+
+            // we generate random but stable guid to allow creating several replication channels from this source to the same destination
+            // so we can start a new bucket migration while the previous one is might be still open
+            var r = new Random((int)BucketMigrationNode.MigrationIndex);
+            Span<byte> guid = stackalloc byte[16];
+            r.NextBytes(guid);
```

Comment:
I checked, there are no dups in the first 10M range using this.

Reply:
Nice, haven't thought about checking this :)